### DISCUSSION TITLE
Installing PiGallery2 on Apache

### DISCUSSION THREAD

**Discussion by wizgod** [bpatrik/pigallery2#370] [Q&A] [answered]
Greetings Programs!

I am using Apache for my web server (managed via Virtualmin).

I am using WebStorm to build and run locally, which works great, but I can't seem to find any instructions on how to run it on the web server.

With my other projects, I just copy the build files in dist to the server, but it would seem I need to follow a different process for PiGallery2.

Any help would be appreciated.

Thanks,

Wg

**Comment by bpatrik** [comment, marked as answer]
I'm not that familiar with this setup. My guess would be that you use Apache to server static files  or php, while this app uses nodejs. You need to run the nodejs server itself.

**Reply by wizgod** [reply]
Hi Patrik,

Firstly, the gallery is awesome as it is exactly what I was looking for and perfect that I can customize it for my needs.

Please forgive the novice questions as I have no experience with docker:

1. I have made changes to the source for my own requirements and after creating a release version with "npm run create-release", what do I do next? Which release files do I move to the server, where? I control the server so I can do whatever is needed.
2. What do I need to do/configure to access the gallery via mydomain.com?

Thank you,

Wg

**Reply by bpatrik** [reply]
Hi sorry for the late reply.

If I remember correctly  `nom run create-release` creates a zip file that contains everything. You need the content of the zip file on your server.
On your server you can directly run the compiled code with `npm start` (given you have node installed on your server. The server should start on port 80. You need to make sure that you are able to access your server's port 80 through mydomain.com (like doing port forwarding on your router, making sure that the domain points to the right ip, etc..)

**Reply by wizgod** [reply]
Hi Patrick,

Thank you, I got it running - I copied over the files in the archive to the server, changed the port to 8000, did an npm install and npm start and configured the proxy on the site in apache.

Is there a way that the gallery can be viewed by guests without having to log in?

Thanks,

Wg
